python/pyproject.toml

```diff
@@ -112,7 +112,7 @@ classifiers = [
 test = [
     "dask-glm==0.3.0",
     "dask-ml",
-    "hdbscan @ git+https://github.com/scikit-learn-contrib/hdbscan.git@master",
+    "hdbscan==0.8.36",
```

Comment:
Upstream released [0.8.38]( https://github.com/scikit-learn-contrib/hdbscan/releases/tag/release-0.8.38 ). This includes a few fixes that we need. Let's switch to that

```suggestion
    "hdbscan==0.8.38",
```